plugins/commands/publish.go

```diff
@@ -175,28 +178,91 @@ func verifyUniqueVersion(pluginName, pluginVersion string, rtDetails *config.Ser
 }
 
 func uploadPlugin(pluginLocalPath, pluginName, pluginVersion, arc string, rtDetails *config.ServerDetails) error {
-	targetPath := utils.GetPluginPathInArtifactory(pluginName, pluginVersion, arc)
-	log.Info("Upload plugin to: " + targetPath + "...")
+	pluginDirRtPath := utils.GetPluginDirPath(pluginName, pluginVersion, arc)
+	log.Info("Upload plugin to: " + pluginDirRtPath + "...")
+	// First uploading resources directory (this is the complex part),if succeeded upload the executable file.
```

Comment:
...part),if succeeded upload... --> ...part). If the upload is successful, upload...